### PR TITLE
feat: Send-code-button-should-be-gray-out-with-empty-or-invalid-Email…

### PR DESCRIPTION
**Pull request by sh1luo**
Fix: #110 
Signed-off-by: wasabi <690898835@qq.com>

### CODE REVIEW

**Comment by sh1luo** [comment]
demo link: http://testb.rescld.cn/signup

Have to be aware of is that the regular expression of email comes from the source code of antd(<https://github.com/yiminghe/async-validator/blob/057b0b047f88fac65457bae691d6cb7c6fe48ce1/src/rule/type.ts#L9>)

the regular expression of email comes from [here](https://regexr.com/3c53v)

File: web/src/auth/SignupPage.js [hunk 1]
```diff
@@ -210,14 +212,21 @@ class SignupPage extends React.Component {
...
-                type: 'email',
-                message: i18next.t("signup:The input is not valid Email!"),
-              },
               {
                 required: required,
                 message: i18next.t("signup:Please input your Email!"),
               },
+              {
+                validator: (_, value) =>{
+                  if( (/^\s*$|^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/).test(this.state.email) ) {
```
Comment (hsluoyz) [inline web/src/auth/SignupPage.js:221]: Put it to Setting.js, provide reference in comment

File: web/src/auth/SignupPage.js [hunk 2]
```diff
@@ -292,6 +302,17 @@ class SignupPage extends React.Component {
                 required: required,
                 message: i18next.t("signup:Please input your phone number!"),
               },
+              {
+                validator: (_, value) =>{
+                  if ( (/^\s*$|^[+]*[(]?[0-9]{1,4}[)]?[-\s/0-9]*$/).test(this.state.phone)) {
```
Comment (hsluoyz) [inline web/src/auth/SignupPage.js:307]: Put it to Setting.js, provide reference in comment

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: web/src/auth/SignupPage.js [hunk 3]
```diff
@@ -23,6 +23,7 @@ import {authConfig} from "./Auth";
 import * as ApplicationBackend from "../backend/ApplicationBackend";
 import * as UserBackend from "../backend/UserBackend";
 import {CountDownInput} from "../component/CountDownInput";
+import {EmailRegEx, PhoneRegEx} from "../Setting.js"
```
Comment (hsluoyz) [inline web/src/auth/SignupPage.js:26]: see: `import * as Setting from "../Setting";`

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**